packages/api/src/controllers/stream.js

```
@@ -309,6 +309,78 @@ app.put('/:id/setactive', authMiddleware({}), async (req, res) => {
+    let sanitized = stream
+    delete sanitized.streamKey
+
+    // const all = false // TODO remove hardcoding here 
+    // const limit = 10 // hard limit so we won't spam endpoints, TODO , have a better adjustable limit 
+    // get a list of user defined webhooks
+    // const filter1 = all ? (o) => o : (o) => !o[Object.keys(o)[0]].deleted
+    // let filter2 = (o) => o[Object.keys(o)[0]].userId
+    // const resp = await req.store.list({
+    //   prefix: `webhook/`,
+    //   limit,
+    //   filter: (o) => filter1(o) && filter2(o),
+    // })
+
+    const resp = await req.store.query({
+      kind: "webhook",
+      query: {
+        userId: stream.userId,
+        eventType: "streamStarted",
+        deleted: false
+      }
+    })
+    
+    let output = resp.data
+    res.status(200)
+
+    await Promise.all(
+      output.map(async (webhook) => {
+        let ip = await dns.resolve4(webhook.url)
+        let isLocal = isLocalIP(ip)
+        if (isLocal) {
+          // don't fire this webhook.
+          console.log(`webhook ${webhook.id} resolved to a localIP, url: ${webhook.url}, resolved IP: ${ip}`)
+        } else {
+          // go ahead
+          let params = {
+            method: 'POST',
+            headers: {
+              'content-type': 'application/json',
+              'user-agent': 'livepeer.com'
+            },
+            timeout: 10 * 1000, // 10 second timeout
+            body: {
+              event: 'streamStarted',
+              stream: JSON.stringify(sanitized),
+            }
+          }
+    
+          try {
+            console.log(`webhook ${webhook.id} firing`)
+            let resp = await fetchWithTimeout(webhook.url, params)
+            if (resp.status === 200) {
+              // all is good
+              console.log(`webhook ${webhook.id} fired successfully`)
+            } else {
+              // block this
+              console.log(`webhook ${webhook.id} didn't get 200 back! response status: ${resp.status}`)
```

We'll want to throw an error here to block stream start